tasks/src/main/java/ru/sberbank/school/task02/calculator/CurrencyCalculator.java

```diff
@@ -0,0 +1,84 @@
+package ru.sberbank.school.task02.calculator;
+
+import ru.sberbank.school.task02.ExternalQuotesService;
+import ru.sberbank.school.task02.FxConversionService;
+import ru.sberbank.school.task02.util.ClientOperation;
+import ru.sberbank.school.task02.util.ExternalQuotesServiceDemo;
+import ru.sberbank.school.task02.util.Quote;
+import ru.sberbank.school.task02.util.Symbol;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+
+public class CurrencyCalculator implements FxConversionService {
+    private ExternalQuotesService externalQuotesService;
+    private List<Quote> quotes = new ArrayList<>();
+    private BigDecimal amountOfRequest;
+    private Symbol symbolOfRequest;
+
+    public CurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        this.externalQuotesService = externalQuotesService;
+    }
+
+    @Override
+    public BigDecimal convert(@NonNull ClientOperation operation,
+                              @NonNull Symbol symbol,
+                              @NonNull BigDecimal amount) {
+
+//        externalQuotesService = new ExternalQuotesServiceDemo();
+        quotes = externalQuotesService.getQuotes(Symbol.USD_RUB);
+        this.amountOfRequest = amount;
+        this.symbolOfRequest = symbol;
+
+        if (check())
```

Comment:
&#39;if&#39; construct must use &#39;{}&#39;s.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a255803480>*